components/esp_http_client/esp_http_client.c

```diff
@@ -690,6 +691,11 @@ static int esp_http_client_get_data(esp_http_client_handle_t client)
     if (client->state < HTTP_STATE_RES_COMPLETE_HEADER) {
         return ESP_FAIL;
     }
+
+    if (client->connection_info.method == HTTP_METHOD_HEAD) {
+        return 0;
```

Comment:
0 means ESP_OK? It would be better to use more meaningful definition instead of 0.

Reply:
No it means 0 bytes read. The return of the function is the number of bytes read. The above `ESP_FAIL` return seems to rely on the fact that currently `ESP_FAIL` is `#define`'d to `-1`.  I wavered in putting the `HEAD` check below in `esp_http_client_perform` and just skipping looking at the body at all, but then the state check above would need to move as well, and the correctly data length for a `HEAD` request is `0` since the RFC states that it "must not return a message-body in the response", so the `esp_http_client_get_data` seemed more appropriate.